### ISSUE TITLE
Rest API

### ISSUE THREAD

**Issue by jhitchcock** [serverfireteam/panel#166]
Hey, 
I'm wanting to add a REST API to this. Has anyone done this already, or have any advice?

**Comment by serverfire** [comment]
@jhitchcock that is great idea , we could help you with that , let's talk about it

**Comment by jhitchcock** [comment]
The actual API part should be pretty straightforward.. But It will also need a layer of users/roles and authentication such as api tokens .. That's the bit I'd struggle with I guess

**Comment by lagbox** [comment]
Luckily with 5.2 there is a token guard that can be used for stateless auth instead of the default session guard.

**Comment by serverfire** [comment]
@jhitchcock  we could keep it simple first , a controller where you can create api key and secret ( we have that on other project we have done ) a route to let users access crud ( update , insert , delete , list ) . what you think ?

**Comment by serverfire** [comment]
@lagbox  Yes we can use that , thank you , would be great if you help on this too :)

**Comment by serverfire** [comment]
@jhitchcock  any news on it ?

**Comment by jhitchcock** [comment]
Howdy, Unfortunately due to time constraints I had to build this fast so I ended up writing a custom api controlller.. Hacking it in basically ;) 

However that has enabled me to become more familiar with the code of the panel, so I'm probably more geared up to write this properly now! 

I'll make a start on it next week, if we go with what you put in your ticket above, for basic public (none user secure API) then it should be fairly quick. 

For my proof of concept I created the Laravel Users table etc seperate to admins, so offered up a Registration and Login method on an authenticate API, and used jwt to supply a token after login. Something like this we can add after as it got a bit messy when I started segeragating table information by user!
